pydantic/networks.py

```diff
@@ -118,13 +287,160 @@ def __get_pydantic_core_schema__(cls, source: type[Any], handler: GetCoreSchemaH
             return schema
 
 
-class _BaseMultiHostUrl(MultiHostUrl):
+class _BaseMultiHostUrl:
     _constraints: ClassVar[UrlConstraints] = UrlConstraints()
+    _url: _CoreMultiHostUrl
+
+    @cached_property
+    def _validator(self) -> TypeAdapter:
+        return TypeAdapter(self.__class__)
```

Comment:
It feels like it could be both simpler and more efficient if here this `TypeAdapter` just built the underlying rust `Url`, with appropriate constraints? Is there a reason why that might be problematic for subclasses? 🤔

Reply:
Hmm, I don't think there's an easy way to do this - you can't easily create a `TypeAdapter` with just a core schema?

Reply:
well we could use a `SchemaValidator` instead of a `TypeAdapter`?